errors_map.go

```diff
@@ -0,0 +1,15 @@
+// Generated by go generate; DO NOT EDIT.
+
+package main
+
+var KnownErrors = map[string]error {
```

Comment:
Feels like we should make this smarter with an init() function and possibly using reflect to fill in the table. Thoughts?

Reply:
This table is filled in by `go generate`, running the code in `gen_errors_maps.go`. So, whenever a new error is added to `errors.go`, we need to run `go generate` to update this. I will add a CI job to validate that these two are in sync.

Reply:
I thought of using `reflect`, too, but this seems more straightforward (and incurs less runtime overhead, which would still be negligible).